src/bootman/sysconfig.c

```diff
@@ -86,16 +86,67 @@ int cbm_get_fstype(const char *boot_device)
         return ret;
 }
 
+static void cmb_inspect_root_native(SystemConfig *c, char *realp) {
+        bool native_uefi = false;
+        autofree(char) *fw_path;
+        char *boot = NULL;
+
+        /* typically /sys, but we forcibly fail this with our tests */
+        fw_path = string_printf("%s/firmware/efi", cbm_system_get_sysfs_path());
+        native_uefi = nc_file_exists(fw_path);
+
+        /* Try to find the system ESP */
+        if (native_uefi && !getenv("CBM_FORCE_LEGACY")) {
```

Comment:
This is a little bit trickier of a case. Do we support systems having a valid EFI and to have been installed with a legacy bootloader? It seems like we'd just have installed with our normal EFI method in that case. Then on systems without efi it would just attempt to get the legacy device instead.

Reply:
This also brings to my attention that you'd need a mechanism to get the env setting into whatever (the swupd update service, the end user shell, etc) is running cbm in a consistent way which seems like a potential place for inconsistent behavior.

Reply:
We don't want "end users" to manually instruct the install process to "force legacy". I should have left a note stating this: "this use of CBM_FORCE_LEGACY on native case must be dropped whenever we cut a new release", with that we'll have an installer image with the latest c-b-m capable of installing clr on non UEFI systems.

In other words this CBM_FORCE_LEGACY condition is only useful for development, we want it all to be auto-detected.

It's not the same for image mode, in that case we want clr-installer to set CBM_FORCE_LEGACY when legacyBios is set to true on yaml files when using the massinstaller.

Reply:
Ah that clears things up for me. That sounds perfect.